Dangerous chemical fixes, chemistry tweaks and improvements to game readability.

Kell-E opened the pull request:
Chemistry suffers from a lot of nonfunctional chemicals, most of which are of the dangerous variety. This PR seeks to rebalance and breathe some life into them, along with fixing and tweaking a few minor things along the way.

This is also my first PR, I am so super sorry if I did anything wrong. I'll try to correct as much as I can but bear with me, I'm not used to this stuff at all!

- Amatoxin is now a delayed activation toxin, incredibly lethal but very slow to activate. It works by creating a second chemical 'Amaspores' which remain dormant in the body, once the amatoxin has run it's course the amaspores violently begin to attack the body at a rapid rate. 
- Lexorin is now much faster to act and is skin permeable but remains easily mitigated by dexalin. While the chemical is very similar to it's previous incarnation it now directly absorbs oxygen from the blood making it a potent threat rather than a mild inconvenience. May require some minor tweaks to death sting to compensate for the massively increased lethality, but the effects of lexorin quickly pass once the chemical is removed from the blood. 
- Carpotoxin properly earns the title of neurotoxin, a few drops can cause massive brain damage via seizures. While it's exceptionally dangerous and liable to kill without help, it does little to prevent the victim from calling for help and it's also exceptionally difficult to obtain. There may be an argument from removing this from the random toxin vial in the uplink, given that it's also used in the creation of another drug but that's somewhat outside of the scope of this PR.
- Impedrezene is exceptionally debilitating and will have lasting consequences for ingesting but takes an overwhelming dosage to become lethal. An excellent drug for use in situations when you want to keep someone under control, doesn't require much to be effective but will still leave the victim with 70%~ of their brain activity in most cases. 

- The smoke chemical now scales with the amount of reagents used to create it, 60u of reagents will now create smoke for 12 seconds before adding in a random factor of 1.5 to 3 seconds. The amount splashed per cloud was potentially excessive given the new duration of smoke so a reduction by 50% (10u to 5u) to the maximum amount of chemicals splashed was in order.

- Lexorin and Acid now present obvious warning messages to both the victim and nearby characters to make it obvious that the chemical being sprayed is in fact, dangerous. It's only fair that chemicals that work on touch inform people, weapons and dangerous things of this caliber should be obvious once deployed. Lexorin no longer works on vox due to their unusual biology.

- The message for being sprayed with acid while wearing protective gear has been obfuscated, you no longer instantly know that the wet stuff was acid because your glasses are resistant.



All of the toxins mentioned above were exceptionally weak, if to the point of being non-functional. Amatoxin took the better part of an hour to kill someone, Carpotoxin and impedrezene were both out healed by natural regeneration and lexorin was largely neutered by natural healing and the oxygen deprivation was so small as to not matter except in dosages close to one hundred units.

Chemical smoke essentially did nothing, the cloud patterns were too unreliable for smoke that lasted for such a short duration that often the first clouds were vanishing before the last ones had even reached their positions. Increasing the timer to be in line with smoke grenades should massively increase the potential for usage.

Acid having no messages for the horrific damage it could cause could often mean several people would sit and watch confused as someone had all of their limbs turn to ash. No more confused staring!

:cl: Yvesza
balance: Improved the effects of Amatoxin, Lexorin, Carpotoxin and Impedrezene. 
balance: Vox are no longer effected by Lexorin.
balance: Chemical smoke duration now scales with the amount of reagents used to create it.
balance: Chemical smoke splashes less reagents on touch (10 -> 5)
tweak: Characters being coated with harmful chemicals will receive warning messages, along with anyone close enough to bear witness.
tweak: Being sprayed with acid while wearing protective gear no longer automatically identifies it as acid.
/:cl:

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -315,11 +315,16 @@
 /datum/reagent/acid/affect_touch(var/mob/living/carbon/M, var/alien, var/removed) // This is the most interesting
+
+// Gives a little readability to the fact that someone is being sprayed with skin melting acid
+	to_chat(M, "<span class='warning'>Your skin sizzles and burns!.</span>")
```
```suggestion
	to_chat(M, SPAN_DANGER("Your skin sizzles and burns!."))
```

Since the player is actively being damaged here, danger is probably better than warning.

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -315,11 +315,16 @@
+
+// Gives a little readability to the fact that someone is being sprayed with skin melting acid
+	to_chat(M, "<span class='warning'>Your skin sizzles and burns!.</span>")
+	M.visible_message("<span class='warning'>\The [M]'s skin sizzles and burns on contact with the liquid!</span>")
```
```suggestion
	M.visible_message(SPAN_WARNING("\The [M]'s skin sizzles and burns on contact with the liquid!"))
```

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -315,11 +315,16 @@
 		if(H.head)
 			if(H.head.unacidable)
-				to_chat(H, "<span class='danger'>Your [H.head] protects you from the acid.</span>")
+				to_chat(H, "<span class='danger'>Your [H.head] protects you from the liquid.</span>")
```
```suggestion
				to_chat(H, SPAN_WARNING("Your [H.head] protects you from the liquid."))
```
Since there's no damage (you were protected after all), best to just warn the player they were splashed.

Kell-E replied:
Thanks! I wasn't aware of the SPAN_WARNING / SPAN_DANGER stuff,

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -347,7 +352,7 @@
+				to_chat(H, "<span class='danger'>Your [H.glasses] partially protect you from the liquid!</span>")
 				removed /= 2
 			else if(removed > meltdose)
 				to_chat(H, "<span class='danger'>Your [H.glasses] melt away!</span>")
```
```suggestion
				to_chat(H, SPAN_DANGER("Your [H.glasses] melt away!"))
```
Housekeeping

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -347,7 +352,7 @@
 		if(H.glasses)
 			if(H.glasses.unacidable)
-				to_chat(H, "<span class='danger'>Your [H.glasses] partially protect you from the acid!</span>")
+				to_chat(H, "<span class='danger'>Your [H.glasses] partially protect you from the liquid!</span>")
```
```suggestion
				to_chat(H, SPAN_WARNING("Your [H.glasses] partially protect you from the liquid!"))
```
Same as above, player wasn't damaged or in danger of being damaged.

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -333,7 +338,7 @@
 		if(H.wear_mask)
 			if(H.wear_mask.unacidable)
-				to_chat(H, "<span class='danger'>Your [H.wear_mask] protects you from the acid.</span>")
+				to_chat(H, "<span class='danger'>Your [H.wear_mask] protects you from the liquid.</span>")
```
```suggestion
				to_chat(H, SPAN_WARNING("Your [H.wear_mask] protects you from the liquid."))
```

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -59,23 +59,64 @@
+
+/datum/reagent/toxin/amaspores
+	name = "Amaspores"
+	description = "The secndary component to amatoxin poisoning, remaining dormant for a time before causing rapid organ and tissue decay."
```
```suggestion
	description = "The secondary component to amatoxin poisoning, remaining dormant for a time before causing rapid organ and tissue decay."
```

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -328,26 +369,49 @@
-		if(M.losebreath < 22.5)
+		M.take_organ_damage(8 * removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
+		if (prob(10))
+			to_chat(M, "<span class='warning'>Your skin fizzles and flakes away!.</span>")
```
```suggestion
			to_chat(M, SPAN_DANGER("Your skin fizzles and flakes away!."))
```
This should be concerning, best to convey the danger.

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -328,26 +369,49 @@
-		if(M.losebreath < 15)
+		M.take_organ_damage(10 * removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
+		if (prob(10))
+			to_chat(M, "<span class='warning'>Your skin fizzles and flakes away!.</span>")
```
```suggestion
			to_chat(M, SPAN_DANGER("Your skin fizzles and flakes away!."))
```

Generalcamo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -328,26 +369,49 @@
+	touch_met = volume // immediately permiates the skin, also avoids bugs with chemical duplication.
+
+	// The warning messages should only display once per splash, all of the chemicals upon the skin should be absorbed in one tick.
+	to_chat(M, "<span class='warning'>You feel a painful fizzling and your skin begins to flake!.</span>")
```
```suggestion
	to_chat(M, SPAN_DANGER("You feel a painful fizzling and your skin begins to flake!."))
```

Generalcamo commented:
Changelog is also formatted incorrectly; tags should be seperated with `;`, not `-`

SierraKomodo commented on code/game/objects/effects/chem/chemsmoke.dm:
```diff
@@ -223,7 +223,7 @@
 		smoke = passed_smoke
 	else
-		smoke = new /obj/effect/effect/smoke/chem(location, smoke_duration + rand(0, 20), T, I)
+		smoke = new /obj/effect/effect/smoke/chem(location, smoke_duration + rand(175, 225), T, I) // Smoke lasts between seventen and a half seconds and twenty two and a half seconds
```
```suggestion
		smoke = new /obj/effect/effect/smoke/chem(location, smoke_duration + rand(17.5 SECONDS, 22.5 SECONDS), T, I)
```

Kell-E replied:
I've re-written the way that smoke duration is calculated based on feedback, 60u of the smoke reagent equates to 12 seconds before adding a random factor the duration. This should scale to allow for stronger or weaker grenades, but may require modification of already existing grenade types that used smoke reagent (Notably tear gas).

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -315,11 +315,16 @@
 
 /datum/reagent/acid/affect_touch(var/mob/living/carbon/M, var/alien, var/removed) // This is the most interesting
+
+// Gives a little readability to the fact that someone is being sprayed with skin melting acid
```
Comment's not really necessary - It's fairly apparent what the message is.

```suggestion
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Dispenser.dm:
```diff
@@ -315,11 +315,16 @@
 /datum/reagent/acid/affect_touch(var/mob/living/carbon/M, var/alien, var/removed) // This is the most interesting
+
+// Gives a little readability to the fact that someone is being sprayed with skin melting acid
+	to_chat(M, "<span class='warning'>Your skin sizzles and burns!.</span>")
+	M.visible_message("<span class='warning'>\The [M]'s skin sizzles and burns on contact with the liquid!</span>")
```
`visible_message()` will also appear for `M`, which makes `to_chat()` redundant. `visible_message()` also accepts a second parameter that is displayed to `M` instead of the message that's displayed to everyone else, so:
```suggestion
	M.visible_message(
		SPAN_WARNING("\The [M]'s skin sizzles and burns on contact with the liquid!</span>"),
		SPAN_DANGER("Your skin sizzles and burns!")
	)
```

Kell-E replied:
I wasn't aware of the second parameter, I've amended where relevant. Thank you!

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -59,23 +59,64 @@
 	strength = 10
 
-/datum/reagent/toxin/carpotoxin
+/datum/reagent/toxin/amatoxin/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
```
`var/` is redundant in proc parameters.

```suggestion
/datum/reagent/toxin/amatoxin/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -59,23 +59,64 @@
+	color = "#330e00"
+	strength = 30
+
+/datum/reagent/toxin/amaspores/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
```
```suggestion
/datum/reagent/toxin/amaspores/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -59,23 +59,64 @@
 	target_organ = BP_BRAIN
 	strength = 10
 
+/datum/reagent/toxin/carpotoxin/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
```
```suggestion
/datum/reagent/toxin/carpotoxin/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -328,26 +369,49 @@
 
 /datum/reagent/lexorin/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
-	if(alien == IS_DIONA)
+	if(alien == IS_DIONA || alien == IS_VOX) // Lexorin now focuses on removing oxygen from the blood, it wouldn't make sense that these two races are affected
```
Vox being immune to lexorin should be documented in the changelog.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -328,26 +369,49 @@
 			M.losebreath++
+	M.adjustOxyLoss(15 * removed)
+
+/datum/reagent/lexorin/affect_touch(var/mob/living/carbon/M, var/alien, var/removed)
```
```suggestion
/datum/reagent/lexorin/affect_touch(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -328,26 +369,49 @@
+	touch_met = volume // immediately permiates the skin, also avoids bugs with chemical duplication.
+
+	// The warning messages should only display once per splash, all of the chemicals upon the skin should be absorbed in one tick.
+	to_chat(M, "<span class='warning'>You feel a painful fizzling and your skin begins to flake!.</span>")
+	M.visible_message("<span class='warning'>\The [M]'s skin fizzles and flakes on contact with the liquid!</span>")
```
See above about `to_chat()` vs `visible_message()`

SierraKomodo reviewed (commented):
Also, about the smoke time changes. I'm not sure all smoke reactions should be that long. There may be cases where the shorter time is desired, whether it be for balancing purposes, for weaker grenades, or for other reasons a player may specifically want - A chemical or other way to modify time, such as based on volume, may be a better option.

MuckerMayhem commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -59,23 +59,64 @@
+	if(prob(20))
+		M.confused += 10
+
+	M.adjustBrainLoss(20 *removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
```
```suggestion
	M.adjustBrainLoss(20 * removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
```

MuckerMayhem commented on code/game/objects/effects/chem/chemsmoke.dm:
```diff
@@ -223,7 +223,7 @@
 		smoke = passed_smoke
 	else
-		smoke = new /obj/effect/effect/smoke/chem(location, smoke_duration + rand(0, 20), T, I)
+		smoke = new /obj/effect/effect/smoke/chem(location, smoke_duration + rand(175, 225), T, I) // Smoke lasts between seventen and a half seconds and twenty two and a half seconds
```
Didn't show Sierra's identical review when I was doing this for whatever reason. whoops.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -68,19 +68,19 @@
 	strength = 10
 
-/datum/reagent/toxin/amatoxin/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+/datum/reagent/toxin/amatoxin/affect_blood(var/mob/living/carbon/M, alien, removed)
```
```suggestion
/datum/reagent/toxin/amatoxin/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -68,19 +68,19 @@
 	strength = 30
 
-/datum/reagent/toxin/amaspores/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+/datum/reagent/toxin/amaspores/affect_blood(var/mob/living/carbon/M, alien, removed)
```
```suggestion
/datum/reagent/toxin/amaspores/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -106,15 +106,15 @@
 	strength = 10
 
-/datum/reagent/toxin/carpotoxin/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+/datum/reagent/toxin/carpotoxin/affect_blood(var/mob/living/carbon/M, alien, removed)
```
```suggestion
/datum/reagent/toxin/carpotoxin/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -379,32 +379,34 @@
 	value = 2.4
 
-/datum/reagent/lexorin/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+/datum/reagent/lexorin/affect_blood(var/mob/living/carbon/M, alien, removed)
```
```suggestion
/datum/reagent/lexorin/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -379,32 +379,34 @@
 	M.adjustOxyLoss(15 * removed)
 
-/datum/reagent/lexorin/affect_touch(var/mob/living/carbon/M, var/alien, var/removed)
+/datum/reagent/lexorin/affect_touch(var/mob/living/carbon/M, alien, removed)
```
```suggestion
/datum/reagent/lexorin/affect_touch(mob/living/carbon/M, alien, removed)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -659,7 +661,7 @@
 	value = 1.8
 
-/datum/reagent/impedrezene/affect_blood(var/mob/living/carbon/M, var/alien, var/removed)
+/datum/reagent/impedrezene/affect_blood(var/mob/living/carbon/M, alien, removed)
```
```suggestion
/datum/reagent/impedrezene/affect_blood(mob/living/carbon/M, alien, removed)
```

SierraKomodo reviewed (changes requested):
Still need to do some testing with the current smoke settings as well.

SierraKomodo reviewed (commented):
Did some testing and experimentation locally and compared to live. This does have a moderate affect on balance regarding both the smoke and amatoxin changes, so I'll be poking the other devs for their opinions as well. Here's my take:

- Teargas grenades are definitely affected by the smoke changes. They exist for roughly 20 seconds with the changes, as opposed to dissipating nearly immediately after exploding on the current dev. Personally, I think this may actually be a good change, or at least one worth trying as it turns them into area denial, and should discourage security from using them for anti-hostage assaults due to the extended exposure. (Flashbangs already serve a similar 'quick use disabler' role anyway)
- Standard smoke with no other chemicals still causes coughing and debilitates you, which may hinder the intention of using it for concealment. Might be worth looking at in the future.
- The dialog for spawning reagent containers sucks because it has no semblemce of sorting or searching functionality and I should really update that at some point.
- The amatoxin/amaspores interaction is really powerful and scales really fast - A single unit of amatoxin results in about 10 amaspores, which results in an even higher amount of amaspores as it appears to create more of itself, which was enough to cause minor liver damage. 5 units of amatoxin produces enough amaspores (20u in blood, 60u metabolised by the time it became 'active') to turn the liver to decaying within only a minute or two of the spores becoming active, and then proceeded to severely damage the kidneys. A full 15 unit injection would probably be lethal within minutes of becoming active, with no apparent means of being aware it's even going to happen - Not even with a scan, as scanners won't show amatoxin or amaspores normally.
- Amaspores appear to not become active until all metabolized amatoxin is gone, not just when the reagents are clear from the bloodstream. Unsure if intentional or not - This doesn't seem to result in more amaspores producing, simply increases the delay before they become active.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -59,23 +59,64 @@
 	description = "A deadly neurotoxin produced by the dreaded space carp."
 	taste_description = "fish"
 	reagent_state = LIQUID
+	metabolism = REM / 4
```
This is redundant - The parent `/datum/reagent/toxin` type is already set to this.
![Code_2021-06-25_08-29-46](https://user-images.githubusercontent.com/11140088/123448219-85fe3080-d58f-11eb-9b10-432240824cf5.png)
```suggestion
```

Kell-E replied:
Probably also the reason why it refused to metabolize properly, thanks!

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -59,23 +59,64 @@
 	target_organ = BP_BRAIN
 	strength = 10
 
+/datum/reagent/toxin/carpotoxin/affect_blood(mob/living/carbon/M, alien, removed)
+	if (prob(3))
+		M.seizure()
+		M.adjustBrainLoss(rand(12, 24))
+
+	if(prob(20))
+		M.confused += 10
+
+	M.adjustBrainLoss(20 * removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
+	M.slurring = max(M.slurring, 10)
+
```
This needs the diona species check to remain in line with other toxins unless the species maintainer approves of this affecting them.
```suggestion
/datum/reagent/toxin/carpotoxin/affect_blood(mob/living/carbon/M, alien, removed)
	if (alien == IS_DIONA)
		return

	if (prob(3))
		M.seizure()
		M.adjustBrainLoss(rand(12, 24))

	if(prob(20))
		M.confused += 10

	M.adjustBrainLoss(20 * removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
	M.slurring = max(M.slurring, 10)

```

Kell-E replied:
It has the diona check?

Kell-E replied:
Oh my god, no it doesn't. My mistake!

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -59,23 +59,64 @@
+	if(prob(20))
+		M.confused += 10
+
+	M.adjustBrainLoss(20 * removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
```
This seems to be resulting in carpotoxin having no affect above 10u, and limited effect until the amount in the blood stream reduces. I suspect `* removed` is doing the opposite of the intent.

Additionally, as per discord, directly damaging the brain is _extremely_ dangerous and lethal. Suggestion would be to try bumping damage dealt through the normal toxin avenues, or target another organ instead.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -328,26 +369,51 @@
-		if(M.losebreath < 22.5)
+		M.take_organ_damage(8 * removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
+		if (prob(10))
+			to_chat(M, SPAN_DANGER("Your skin fizzles and flakes away!."))
```
Making this a visible message can make it more apparent what's happening for medical.
```suggestion
			M.visible_message(
				SPAN_WARNING("\The [M]'s skin fizzles and flakes away!"),
				SPAN_DANGER("Your skin fizzles and flakes away!")
			)
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -328,26 +369,51 @@
-		if(M.losebreath < 15)
+		M.take_organ_damage(10 * removed, 0, ORGAN_DAMAGE_FLESH_ONLY)
+		if (prob(10))
+			to_chat(M, SPAN_DANGER("Your skin fizzles and flakes away!."))
```
```suggestion
			M.visible_message(
				SPAN_WARNING("\The [M]'s skin fizzles and flakes away!"),
				SPAN_DANGER("Your skin fizzles and flakes away!")
			)
```

SierraKomodo reviewed (changes requested):
Aside from the visible_message(), the lexorin changes look good to me - Lethality's increased without being too lethal and still being saveable as long as people act relatively quickly.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -595,16 +661,25 @@
+
 	if(prob(80))
-		M.adjustBrainLoss(5.25 * removed)
+		M.confused += 5
```
This needs an upper limit check to avoid the potential of indefinite confusion.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm:
```diff
@@ -595,16 +661,25 @@
 		M.emote("drool")
+		M.apply_effect(STUTTER, 3)
+
+	if (M.getBrainLoss() < 60)
+		M.adjustBrainLoss(20*removed)
+	if (M.getBrainLoss() >= 60)
+		M.adjustBrainLoss(5*removed)
```
As a note, this results in impedrezine dropping brain activity down to around 60-70% with a 15u dosage. Prior to this change, brain activity seemed to mostly stay above 95% with the same dosage, if it ever dropped at all.

This seems contradictory compared the the comment that was added to impedrezine.

Kell-E replied:
Changed the values a little, a 5u dose brings someone down to 80% brain activity. A 30u dose brings someone down to 49% activity. It'd take around 90u to kill someone, I actually had to turn the damage values up for the post threshold damage in order for this to beat out natural regeneration.

SierraKomodo commented:
Okay that's all the stuff tested and reviewed now.

Kell-E commented:
Pushed further changes, altered the effect of carpotoxin to no longer immediately attack the brain. Along with reducing the effectiveness of amatoxin (and giving it some minor warnings and effects similar to a minor allergic reaction to give some idea that something is wrong), along with some minor fixes to impedrezene so that it'll function correctly (Too fast to act initially and then was all together out healed by natural regeneration, which it'll now ever so slightly beat out._

SierraKomodo reviewed (approved):
Both carpotoxin and amatoxin changes look good in testing - Amatoxin's potentially lethal, but delayed and has warning signs before it goes all out.

The chemistry wiki will probably need updating for any relevant changes.

SierraKomodo reviewed (changes requested):
Needs squash before final approval actually.

SierraKomodo commented:
Also, updated wiki prototype here: https://wiki.baystation12.net/User:Sierrakomodo/List_of_Miscellaneous_Chemicals_PR_30855

Kell-E commented:
Okay! I'll see what I can do about a squash, who would be the best person to ask about getting a wiki account? Updating the page to reflect recent changes shouldn't be too much of a big deal.

SierraKomodo commented:
> Okay! I'll see what I can do about a squash, who would be the best person to ask about getting a wiki account? Updating the page to reflect recent changes shouldn't be too much of a big deal.

Eckles is your best bet on the discord, or you can just let me know what changes to make and I'll apply them to the prototype. I've already adjusted the description for carpotoxin as a note.

Kell-E commented:
Thank you, I'll get updated descriptions for the changed chemicals written up shortly.

Kell-E commented:
Lexorin -> Temporarily stops respiration and reacts with oxygen in the blood to cause tissue damage, extremely dangerous and skin permeable. 
Amatoxin -> A powerful delayed poison, will initially cause a moderate allergic reaction in the host while producing a large dose of inert spores. Once the Amatoxin has run it's course the spores will come to life and cause severe organ damage, large doses are very slow to act but will cause an extreme amount of damage.

Since Amaspores are only a product of Amatoxin and relatively difficult to obtain (Dialysis on someone filled with Amatoxin) I'm not sure it's worth listing them as a chemical, but for the sake of completeness I'll also add include the details required for a new entry on the wiki and I'll leave it up to you if you think it's worth an inclusion.

Amaspores (Colour: #330e00) (Recipe: N/A) -> The secondary product of amatoxin poisoning, so long as amatoxin remains in the system Amaspores will lay dormant. Controlled substance (Yes), Immediate reaction (No), Injectable (Yes)


Impedrezene remains close enough that it probably doesn't require an updated entry.

Kell-E committed:
Dangerous chemical fixes.

Chemistry suffers from a lot of nonfunctional chemicals, most of which are of the dangerous variety.  

- Amatoxin is now a delayed activation toxin, incredibly lethal but very slow to activate.
- Lexorin is now much faster to act and is skin permeable but remains easily mitigated by dexalin.
- Carpotoxin properly earns the title of neurotoxin, a few drops can cause massive brain damage via seizures.
- Impedrezene is exceptionally debilitating and will have lasting consequences for ingesting but takes an overwhelming dosage to become lethal.

- The smoke chemical has had it's duration increased from 1 to 1.5 seconds to 17.5 to 22.5 seconds, based on the 20 second smoke duration present in already existing smoke grenades.

- Lexorin and Acid now present obvious warning messages to both the victim and nearby characters to make it obvious that the chemical being sprayed is in fact, dangerous.

- The message for being sprayed with acid while wearing protective gear has been obfuscated, you no longer instantly know that the wet stuff was acid because your glasses are resistant.


All of the toxins mentioned above were exceptionally weak, if to the point of being non-functional. Amatoxin took the better part of an hour to kill someone, Carpotoxin and impedrezene were both out healed by natural regeneration and lexorin was largely neutered by natural healing and the oxygen deprivation was so small as to not matter except in dosages close to one hundred units.

Chemical smoke essentially did nothing, the cloud patterns were too unreliable for smoke that lasted for such a short duration that often the first clouds were vanishing before the last ones had even reached their positions. Increasing the timer to be in line with smoke grenades should massively increase the potential for usage.

Acid having no messages for the horrific damage it could cause could often mean several people would sit and watch confused as someone had all of their limbs turn to ash. No more confused staring!
Quick bugfix for scannable lexorin

Forgot to reset my testing things!
Code changes based upon review

Several changes, most notably to chemical smoke duration scaling. Several small changes to the way that displaying of messages was handled to prevent duplication of messages.
Further changes based upon review


Further changes based upon review

Reduced the effectiveness of amatoxin and gave it additional, more obvious messages to imply that something is going wrong. Carpotoxin has been altered to slowly cause clotting and eventually cardiac arrest instead of outright attacking the brain. 
Altered the effectiveness of Impedrezene to be slower but buffed the post threshold damage slightly so that it can beat out natural regeneration ever so slightly.
